### PR TITLE
Scraping the100.io game gamertags from game link.

### PR DESCRIPTION
**Pull request by michael-donat**
Summary of changes:

The Gamertag input is now an 'omnisearch' - it will recognise the100.io game link (`https://www.the100.io/game/967740`), regex match the game id and run a scrape against the100.io to get all gamertags. Once those are received it will add them in a loop reusing the `vm.add()` method that already existed.

One annoying factor is that because of cors and mime checking settings, there is no way to directly scrape the100.io from the browser, a server side component is needed.

This component is provided in `server/index.js` and has it's own `package.json`, will need to be installed and then ran (preferably using something like nodemon or pm2). By default it will bind itself to `:3000`

There's also been a change to cofig, the `apiKey.json` file now contains the100endpoint to look for a game (this is mostly for development, I will explain that below).

#### So, how to deploy it all together?

1. Client code / Extension

Nothing changes here, the current README instructions are still applicable, just build dist and chuck it out.

2. Server code

This needs to be put in a place where you want to run it, `npm installed` and then `node run`. By default it will be on port 3000, but you can control it with `process.env.PORT`

You will then need to provide a proxyPass directive to your apache config, by default the client code is expecting to find info about gamertags under `http://destinyraidstatus.com/api/the100/game/:game` so what we want to do is make all requests to `http://destinyraidstatus.com/api/` to proxy to our node server process.

Not sure how you have everything configured, but following should be a good start (assuming you are running that server component on port 3000):

https://httpd.apache.org/docs/2.4/mod/mod_proxy.html

```
ProxyPass "/api" "http://127.0.0.1:3000/"
```

So we are simply looking at proxying a call from the client (`http://destinyraidstatus.com/api/the100/game/123123`) into a call that will go into our server component (`http://localhost:3000/api/the100/game/:game`), ProxyPass should do just this.

#### For development?

Here you just need to run the server component locally and then change apiKey.json and add that endpoint parameter to point to localhost, example below

```
{ "apiKey": "", "the100endpoint": "http://127.0.0.1:3000/api/the100/game/" }
```

That should be all you need.

[example](https://dl.dropboxusercontent.com/content_link/dQszVTP1M6qrhOMq0xl3ICveTxu7wRGZT6qLN29S1SqAxEksgpAtSwiKHksBZjSG/file)

### CODE REVIEW

**Comment by michael-donat** [comment]
Will rebase to get rid of those conflicts.

**Comment by michael-donat** [comment]
Ok, all rebased, up to date, and working.

Lemme know if any changes needed, I've been using it locally already and it all seems fine.

**Comment by michael-donat** [comment]
Actually, to avoid the parodying you could simply define another vhost under api.destinyraidstatus.com and host it there, whatever fits your environment better.

File: app/scripts/drsThe100.service.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+angular
+    .module('drsApp')
+    .factory('The100Service', The100Service);
+
+The100Service.$inject = ['$q', '$http', '$sce'];
+
+var the100endpoint = $DRS_100_ENDPOINT;
+
+function The100Service($q, $http, $sce) {
```
Comment (Mulchman) [inline app/scripts/drsThe100.service.js]: I don't see `$sce` ever being used?

File: app/scripts/drsThe100.service.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        scrapeGamertags: scrapeGamertags,
+    };
+    return service;
+
+
+    function scrapeGamertags(gameId) {
+
+        var url = the100endpoint + gameId;
+
+        const p = $q.defer();
```
Comment (Mulchman) [inline app/scripts/drsThe100.service.js]: What's `p` for? I don't see it being resolved/rejected.

File: app/scripts/drsInputPlayer.directive.js [hunk 3]
```diff
@@ -16,22 +16,49 @@ function InputPlayer() {
...
+  vm.run = run;
   vm.player = "";
   vm.keyup = keyup;
   vm.toggle = toggle;
   vm.platform = getPlatformFromSettings();
 
+    function run() {
+        var input = vm.player;
+
+        var match;
```
Comment (Mulchman) [inline app/scripts/drsInputPlayer.directive.js]: Prefer let/const

**Review by Mulchman — CHANGES_REQUESTED** [review]

**Comment by Mulchman** [comment]
I think this is going to be really cool.

So that proxyPass thing is essentially doing this for us-

`http://www.destinyraidstatus.com/index.html#/api/the100/game/:game`

right?

Would it be easier to just define a route in `drsApp.routes.js`?

I almost want to set HTML5 mode true to allow the removal of the hash bang stuff but it messes with my Chrome extension development workflow.

**Comment by michael-donat** [comment]
@Mulchman no, ProxyPass will do `http://www.destinyraidstatus.com/api/the100/game/:game` the angular app is not invoked. You can do this without proxy pass by defining another vhost and subdomaining it to `http://api.destinyraidstatus.com`

You need to run the node process separately.

**Review by Mulchman — APPROVED** [review]

**Comment by Mulchman** [comment]
I've created `http://api.destinyraidstatus.com`

**Comment by michael-donat** [comment]
I suppose it still needs to propagate ;)

```
 -> nslookup http://api.destinyraidstatus.com
Server:		8.8.8.8
Address:	8.8.8.8#53

** server can't find http://api.destinyraidstatus.com: NXDOMAIN

mdonat@osx-mdonat.local:/Users/mdonat/Development  $
 ->
```

You will need to adjust the default URL in webpack config to `http://api.destinyraidstatus.com/api/the100/game/`

**Comment by Mulchman** [comment]
I'm not 100% sure I did it right. Still working on npm installing and running the server component.

**Comment by michael-donat** [comment]
Lemme know if you need a hand, may be easier to collab via hangouts chat or something. It’s almost 11pm here, but I should stay online for another hour or two, catching up on the playoffs ;)

**Comment by Mulchman** [comment]
Yeah, I think we'll need to chat at some point. I'm not 100% sure how to npm install your code and then run it. What I did results in a "can't find module" error when invoking `npm run .`

I installed `nvm` then installed `node` through that. I then ran `npm install` in the directory your code is in. I then ran `npm run .` to see if anything would happen and that's where I'm at with the error.

I'm a n00b when it comes to server side stuff.

**Comment by michael-donat** [comment]
Just do node index.js instead of npm run .

**Comment by Mulchman** [comment]
How does this process get started and maintained as the server goes up/down? Right now it's just running in my SSH window process.

**Comment by michael-donat** [comment]
It won’t on it’s own.

It’s difficult to propose any solution to this as I know nothing about your environment.

It’s been a long long time since I ran anything behind apache, or without Docker at all.

You could try a module like pm2, should work fine with nvm and will almost configure itself for init.d on it’s own, details here:

http://pm2.keymetrics.io/docs/usage/startup/ <http://pm2.keymetrics.io/docs/usage/startup/>

To start you will need to `npm install -g pm2` and then follow instructions in the link.

Initially you can also just run it by hand as it is by `node index.js &` and test the solition via frontned. There’s no indiation that the input accepts those links so you shouldn’t see users trying to use it.

**Pull request merged**